internal/sdkv2provider/data_source_device_posture_rules_test.go

```diff
@@ -0,0 +1,63 @@
+package sdkv2provider
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
+)
+
+func TestAccCloudflareDevicePostureRulesDataSource(t *testing.T) {
+	rnd := generateRandomResourceName()
+	accountID := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+	name := fmt.Sprintf("data.cloudflare_device_posture_rules.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareDevicePostureRulesConfig(rnd, accountID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, consts.AccountIDSchemaKey, accountID),
+					resource.TestCheckResourceAttr(name, "name", rnd),
+					resource.TestCheckResourceAttrSet(name, "rules.0.id"),
+					resource.TestCheckResourceAttr(name, "rules.0.name", rnd),
+					resource.TestCheckResourceAttr(name, "rules.0.type", "file"),
+					resource.TestCheckResourceAttr(name, "rules.0.description", "check for /dev/random"),
+					resource.TestCheckResourceAttr(name, "rules.0.schedule", "1h"),
+					resource.TestCheckNoResourceAttr(name, "rules.0.expiration"),
```

Comment:
looks like this is failing because the API is returning `null` and that is being stringified as a value.

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareDevicePostureRulesDataSource" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareDevicePostureRulesDataSource
    data_source_device_posture_rules_test.go:18: Step 1/1 error: Check failed: Check 8/8 error: data.cloudflare_device_posture_rules.kvfkelrflt: Attribute 'rules.0.expiration' found when not expected
--- FAIL: TestAccCloudflareDevicePostureRulesDataSource (8.10s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	9.059s
FAIL
make: *** [testacc] Error 1
```

Reply:
It looks like the struct in cloudflare-go isn't a string pointer, and the existing device posture rule resource doesn't handle the null either. Do you think it's ok to just assert it's an empty string?

Reply:
i'm trying to remember the exact go-cty internals here but i _think_ that will do it. let me update the test and make sure that works and if so, roll with that for now and we can fix the struct to handle this better later on.

Reply:
looks like that will do it 👍 

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareDevicePostureRulesDataSource" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareDevicePostureRulesDataSource
--- PASS: TestAccCloudflareDevicePostureRulesDataSource (12.59s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	13.509s
```